File: src/dataset/c/hamming_exercism-fiplox/claude/ClaudeSonnet4_hamming_v4.c
#include "hamming.h"

int compute(const char *lhs, const char *rhs)
{
	int distance = 0;
	const char *p1 = lhs;
	const char *p2 = rhs;
	
	while (*p1 && *p2) {
		if (*p1 != *p2)
			distance++;
		p1++;
		p2++;
	}
	
	if (*p1 || *p2)
		return -1;
	
	return distance;
}

File: src/dataset/python/hamming_exercism-ThomasZumsteg/claude/ClaudeSonnet4_hamming_v4.py
"""Finds the hamming distance between two DNA sequences"""

def distance(strand_a, strand_b):
    """Counts the differences in two sequences of DNA"""
    count = 0
    for i in range(len(strand_a)):
        if strand_a[i] != strand_b[i]:
            count += 1
    return count